### PR TITLE
feat: add rsync spec

### LINKED ISSUE #510: [rsync] completion spec
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [ ] I would like to work on this.

### What CLI tool does this relate to?

rsync

### Which statement makes the most sense?

This completion spec does not exist

### Issue Details

Lots of arguments with the potential to re-use generators from SSH

### PR DESCRIPTION
**Pull request by Guergeiro**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature.

**What is the current behavior? (You can also link to an open issue here)**
`rsync` is not available. Closes #510 .

**What is the new behavior (if this is a feature change)?**
Support for `rsync`.

### CODE REVIEW

File: src/rsync.ts [hunk 1]
```diff
@@ -0,0 +1,831 @@
...
+      description: "Also compare destination files relative to DIR",
+      requiresEquals: true,
+      args: {
+        name: "DIR",
+        template: "folders",
+      },
+    },
+    {
+      name: "--copy-dest",
+      description: "... and include copies of unchanged files",
```
Comment (fedeci) [inline src/rsync.ts:534]: This description seems to be missing something

File: src/rsync.ts [hunk 2]
```diff
@@ -0,0 +1,831 @@
...
+    { name: "--existing", description: "Skip creating new files on receiver" },
+    {
+      name: "--ignore-existing",
+      description: "Skip updating files that already exist on receiver",
+    },
+    {
+      name: "--remove-source-files",
+      description: "Sender removes synchronized files (non-dirs)",
+    },
+    { name: "--del", description: "An alias for --delete-during" },
```
Comment (fedeci) [inline src/rsync.ts:340]: Can you add this one as an array to the --delete-during option?

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks!

File: src/rsync.ts (hunk 2 above)
Comment (Guergeiro) [reply src/rsync.ts:340]: Will do!

File: src/rsync.ts (hunk 1 above)
Comment (Guergeiro) [reply src/rsync.ts:534]: Yup. It's the same as `--compare-dest`. Will add it!

Comment (Guergeiro) [reply src/rsync.ts]: Fixed in c45653a

File: src/rsync.ts (hunk 2 above)
Comment (Guergeiro) [reply src/rsync.ts]: Fixed in c45653a

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**